pkg/stores/proxyalpha/tablelistconvert/client.go

```diff
@@ -0,0 +1,138 @@
+*/
+package tablelistconvert
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/wrangler/pkg/data"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8sWatch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+)
+
+type Client struct {
+	dynamic.ResourceInterface
+}
+
+var _ dynamic.ResourceInterface = (*Client)(nil)
+
+type tableConvertWatch struct {
+	done   chan struct{}
+	events chan k8sWatch.Event
+	k8sWatch.Interface
+}
+
+type WarningBuffer []types.Warning
+
+// List will return an *UnstructuredList that contains Items instead of just using the Object field to store a table as
+// Table Clients do. The items will preserve values for columns in the form of metadata.fields.
+func (c *Client) List(ctx context.Context, opts metav1.ListOptions) (*unstructured.UnstructuredList, error) {
+	list, err := c.ResourceInterface.List(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	tableToList(list)
+	return list, nil
+}
+
+func (c *Client) Watch(ctx context.Context, opts metav1.ListOptions) (k8sWatch.Interface, error) {
+	w, err := c.ResourceInterface.Watch(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	events := make(chan k8sWatch.Event)
+	done := make(chan struct{})
+	eventWatch := &tableConvertWatch{done: done, events: events, Interface: w}
+	eventWatch.feed()
+	return eventWatch, nil
+}
+
+func (w *tableConvertWatch) feed() {
+	tableEvents := w.Interface.ResultChan()
+	go func() {
+		for {
+			select {
+			case e, ok := <-tableEvents:
+				if !ok {
+					fmt.Println("tableEvents closed!")
+					close(w.events)
+					return
+				}
+				if unstr, ok := e.Object.(*unstructured.Unstructured); ok {
+					rowToObject(unstr)
+					w.events <- e
+				}
+			case <-w.done:
+				fmt.Println("TABLE CONVERT CLIENT HAS BEEN CLOSED")
+				close(w.events)
+				return
+			}
+		}
+	}()
+}
+
+func (w *tableConvertWatch) ResultChan() <-chan k8sWatch.Event {
+	return w.events
+}
+
+func (w *tableConvertWatch) Stop() {
+	fmt.Println("stop")
+	close(w.done)
+	w.Interface.Stop()
+}
+
+func rowToObject(obj *unstructured.Unstructured) {
+	if obj == nil {
+		return
+	}
+	if obj.Object["kind"] != "Table" ||
+		(obj.Object["apiVersion"] != "meta.k8s.io/v1" &&
+			obj.Object["apiVersion"] != "meta.k8s.io/v1beta1") {
+		return
+	}
+
+	items := tableToObjects(obj.Object)
+	if len(items) == 1 {
+		obj.Object = items[0].Object
+	}
+}
+
+func tableToList(obj *unstructured.UnstructuredList) {
+	if obj.Object["kind"] != "Table" ||
+		(obj.Object["apiVersion"] != "meta.k8s.io/v1" &&
+			obj.Object["apiVersion"] != "meta.k8s.io/v1beta1") {
+		return
+	}
+
+	obj.Items = tableToObjects(obj.Object)
+}
+
+func tableToObjects(obj map[string]interface{}) []unstructured.Unstructured {
+	var result []unstructured.Unstructured
+
+	rows, _ := obj["rows"].([]interface{})
```

Comment:
I'm using mostly the same conversion logic that we are already using in the proxy store, as mentioned [here](https://github.com/rancher/steve/pull/223#discussion_r1624890534). If for whatever reason it is unavailable I don't think we want to prevent anything from proceeding.